server/coordinator/procedure/ddl/droppartitiontable/drop_partition_table.go

```diff
@@ -361,64 +286,64 @@ func finishCallback(event *fsm.Event) {
 	}
 	log.Info("drop partition table finish")
 
-	tableInfo := cluster.TableInfo{
+	tableInfo := metadata.TableInfo{
 		ID:         request.table.ID,
 		Name:       request.table.Name,
 		SchemaID:   request.table.SchemaID,
-		SchemaName: request.req.GetSchemaName(),
+		SchemaName: request.p.params.SourceReq.GetSchemaName(),
 	}
 
-	if err = request.onSucceeded(tableInfo); err != nil {
+	if err = request.p.params.OnSucceeded(tableInfo); err != nil {
 		procedure.CancelEventWithLog(event, err, "drop partition table on succeeded")
 		return
 	}
 }
 
-func dropTableMetaData(event *fsm.Event, tableName string) (storage.Table, cluster.DropTableResult, bool, error) {
+func dropTableMetaData(event *fsm.Event, tableName string) (storage.Table, metadata.DropTableResult, bool, error) {
```

Comment:
Why are there so many returned values? Shall remove some of them or use a struct to wrap them if they are necessary.